### PR TITLE
LSM upstream changes

### PR DESCRIPTION
**Pull request by xlab**
Picked two PRs to include for the upcoming testnet:
* `#65` - renaming exemption to validator bond
* `#61` - adding cap param and accounting

Better review by individual commits

### CODE REVIEW

File: x/lsnative/staking/keeper/liquid_stake.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+}
+
+// GetTotalLiquidStakedTokens returns the total outstanding tokens owned by a liquid staking provider
+// Returns zero if the total liquid stake amount has not been initialized
+func (k Keeper) GetTotalLiquidStakedTokens(ctx sdk.Context) sdk.Int {
+	store := ctx.KVStore(k.storeKey)
+	tokensBz := store.Get(types.TotalLiquidStakedTokensKey)
+
+	if tokensBz == nil {
+		// QUESTION: Should we panic here instead?
```
Comment (ajeet97) [inline x/lsnative/staking/keeper/liquid_stake.go:28]: If we are adding a migration for this key, then panic makes sense here I guess.

File: x/lsnative/staking/types/keys.go [hunk 2]
```diff
@@ -54,6 +57,7 @@ var (
 	TokenizeShareRecordIdByOwnerPrefix = []byte{0x62} // key for tokenizeshare record id by owner prefix
 	TokenizeShareRecordIdByDenomPrefix = []byte{0x63} // key for tokenizeshare record id by denom prefix
 	LastTokenizeShareRecordIdKey       = []byte{0x64} // key for last tokenize share record id
+	TotalLiquidStakedTokensKey         = []byte{0x65} // key for total liquid staked tokens
```
Comment (ajeet97) [inline x/lsnative/staking/types/keys.go:60]: Should we add migrations for these keys?

**Review by ajeet97 — CHANGES_REQUESTED** [review]

File: x/lsnative/staking/migrations/v046/store.go [hunk 3]
```diff
@@ -22,10 +22,10 @@ func MigrateStore(ctx sdk.Context, storeKey storetypes.StoreKey, cdc codec.Binar
 func migrateParamsStore(ctx sdk.Context, paramstore paramtypes.Subspace) {
 	if paramstore.HasKeyTable() {
 		paramstore.Set(ctx, types.KeyMinCommissionRate, types.DefaultMinCommissionRate)
-		paramstore.Set(ctx, types.KeyExemptionFactor, types.DefaultExemptionFactor)
+		paramstore.Set(ctx, types.KeyValidatorBondFactor, types.DefaultValidatorBondFactor)
```
[the remark this answers is not in the document]
Comment (xlab) [reply x/lsnative/staking/migrations/v046/store.go:30]: Done

File: x/lsnative/staking/keeper/liquid_stake.go (hunk 1 above)
Comment (xlab) [reply x/lsnative/staking/keeper/liquid_stake.go]: Done

File: x/lsnative/staking/types/keys.go (hunk 2 above)
Comment (xlab) [reply x/lsnative/staking/types/keys.go:60]: Added

**Review by ajeet97 — APPROVED** [review]
LGTM

**Review by puneet2019 — APPROVED** [review]

**Pull request merged**